### PR TITLE
Fix some warnings.

### PR DESCRIPTION
**Pull request by KirillOsenkov**

### CODE REVIEW

File: Editor/Completion/XmlSchemaCompletionProvider.cs [hunk 1]
```diff
@@ -136,7 +136,9 @@ public static string GetUri (string fileName)
 
 		public void EnsureLoaded ()
 		{
+#pragma warning disable VSTHRD002 // Avoid problematic synchronous waits
```
Comment (KirillOsenkov) [inline Editor/Completion/XmlSchemaCompletionProvider.cs:139]: should probably use IWaitIndicator

Comment (KirillOsenkov) [reply Editor/Completion/XmlSchemaCompletionProvider.cs:139]: IWaitIndicator is in Microsoft.VisualStudio.Text.Internal.dll, not sure we want to add this dependency?

**Comment by KirillOsenkov** [comment]
Not sure what to do about this one:
https://github.com/mhutch/MonoDevelop.Xml/blob/39af8719c00034518c113faa6815f2bf584c2199/Editor/Commands/CommentUncommentCommandHandler.cs#L79

We could use IWaitIndicator to display a progress bar since we're blocking the UI thread. I doubt there can be a deadlock as the parser doesn't have dependencies on the UI thread.

Or we could use JoinableTaskFactory.

Or just add an ignore for this warning.

**Comment by KirillOsenkov** [comment]
What do you think?

File: Editor/Completion/XmlSchemaCompletionProvider.cs (hunk 1 above)
Comment (mhutch) [reply Editor/Completion/XmlSchemaCompletionProvider.cs:139]: Looks like there are alternatives? https://github.com/dotnet/roslyn/commit/a5019f17d35a8750bfc4f47f826f7d607e128c11

**Comment by mhutch** [comment]
I think I fixed all these in https://github.com/mhutch/MonoDevelop.Xml/pull/54.

In the case you were usure about, I used `joinableTaskContext.Factory.Run` to run the potentially blocking task, and used `commandExecutionContext.OperationContext.UserCancellationToken` to enable cancellation. I _think_ this should work.

Note also that we could probably use `IUIThreadOperationExecutor` in some cases.

**Pull request closed without merging**